### DISCUSSION TITLE
OpenWrt 定时运行 CloudflareSpeedTest 并自动更新到 CF 的域名上

### DISCUSSION THREAD

**Discussion by lee1080** [XIU2/CloudflareSpeedTest#314] [分享]
## 思路
CloudflareSpeedTest是一个非常方便的IP优选工具，但是每次优选完成后还得自己去更新一下ip地址。
所以
写一个shell脚本，让openwrt定时执行。CloudflareSpeedTest获取到最优的IP地址后，调取cf的API将最优的IP更新到域名。访问这个域名就相当于访问这个优选后的IP地址，就非常的方便。
## 项目地址
我将shell脚本托管到GitHub了，有需要的可以配置一个试试。[CloudflareSpeedTestDDNS](https://github.com/lee1080/CloudflareSpeedTestDDNS)

**Comment by sxml** [comment]
这种修改网站DDNS 跟修改防火墙方式 替换地址方式 会反应比较慢吗？替换ip会延迟多久。

**Comment by lee1080** [comment]
> 这种修改网站DDNS 跟修改防火墙方式 替换地址方式 会反应比较慢吗？替换ip会延迟多久。

替换IP的延迟肯定是会有的。从老IP到新IP得看服务商的生效时间，加上本地dns的过期时间。但是也没有想象中那么慢，在替换后，可以在本地设备ping你的域名，看看新IP生效时间有多快。对节点使用来说，使用IP和使用域名延迟是一样的。
